go-controller/pkg/ovn/ipallocator/allocator/bitmap_test.go

```diff
@@ -155,3 +155,173 @@ func TestContiguousAllocation(t *testing.T) {
+	}
+
+	if _, ok, _ := m.AllocateNext(); ok {
+		t.Errorf("unexpected success")
+	}
+}
+
+func TestRoundRobinAllocationOrdering(t *testing.T) {
+	max := 10
+	m := NewRoundRobinAllocationMap(max, "test")
+
+	// Pre-allocate a couple entries at the start of themap
+	for i := 0; i < 3; i++ {
+		if ok, _ := m.Allocate(i); !ok {
+			t.Errorf("error allocate offset %v", i)
+		}
+	}
+
+	// Next allocation should be after the preallocated entries
+	next, ok, _ := m.AllocateNext()
+	if !ok {
+		t.Fatalf("unexpected error")
+	}
+	if next != 3 {
+		t.Fatalf("expect next to 3, but got %d", next)
+	}
+
+	// Release one of the pre-allocated entries
+	if err := m.Release(0); err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+
+	// Next allocation should be after the most recently allocated entry,
+	// not one of the just-released ones
+	next, ok, _ = m.AllocateNext()
+	if !ok {
+		t.Fatalf("unexpected error")
+	}
+	if next != 4 {
+		t.Fatalf("expect next to 4, but got %d", next)
+	}
+}
+
+func TestRoundRobinAllocate(t *testing.T) {
+	max := 10
+	m := NewRoundRobinAllocationMap(max, "test")
+
+	if _, ok, _ := m.AllocateNext(); !ok {
+		t.Fatalf("unexpected error")
+	}
+	if m.count != 1 {
+		t.Errorf("expect to get %d, but got %d", 1, m.count)
+	}
+	if f := m.Free(); f != max-1 {
+		t.Errorf("expect to get %d, but got %d", max-1, f)
+	}
+}
+
+func TestRoundRobinAllocateMax(t *testing.T) {
+	max := 10
+	m := NewRoundRobinAllocationMap(max, "test")
+	for i := 0; i < max; i++ {
+		if _, ok, _ := m.AllocateNext(); !ok {
+			t.Fatalf("unexpected error")
+		}
+	}
+
+	if _, ok, _ := m.AllocateNext(); ok {
+		t.Errorf("unexpected success")
+	}
+	if f := m.Free(); f != 0 {
+		t.Errorf("expect to get %d, but got %d", 0, f)
+	}
+}
+
+func TestRoundRobinAllocateError(t *testing.T) {
+	m := NewRoundRobinAllocationMap(10, "test")
+	if ok, _ := m.Allocate(3); !ok {
+		t.Errorf("error allocate offset %v", 3)
+	}
+	if ok, _ := m.Allocate(3); ok {
+		t.Errorf("unexpected success")
+	}
+}
+
+func TestRoundRobinRelease(t *testing.T) {
+	offset := 3
+	m := NewRoundRobinAllocationMap(10, "test")
+	if ok, _ := m.Allocate(offset); !ok {
+		t.Errorf("error allocate offset %v", offset)
+	}
+
+	if !m.Has(offset) {
+		t.Errorf("expect offset %v allocated", offset)
+	}
+
+	if err := m.Release(offset); err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+
+	if m.Has(offset) {
+		t.Errorf("expect offset %v not allocated", offset)
+	}
+}
+
+func TestRoundRobinForEach(t *testing.T) {
+	testCases := []sets.Int{
+		sets.NewInt(),
+		sets.NewInt(0),
+		sets.NewInt(0, 2, 5, 9),
+		sets.NewInt(0, 1, 2, 3, 4, 5, 6, 7, 8, 9),
+	}
+
+	for i, tc := range testCases {
+		m := NewRoundRobinAllocationMap(10, "test")
+		for offset := range tc {
+			if ok, _ := m.Allocate(offset); !ok {
+				t.Errorf("[%d] error allocate offset %v", i, offset)
+			}
+			if !m.Has(offset) {
+				t.Errorf("[%d] expect offset %v allocated", i, offset)
+			}
+		}
+		calls := sets.NewInt()
+		m.ForEach(func(i int) {
```

Comment:
nit: for clarity, can you change the index variable of the anonymous function to something else other than `i`. Alternatively, change the index variable of the outer for loop for test-cases to something else. It's a bit confusing although it works correctly because of scoping rules.